python/pyarrow/tests/test_compute.py

```diff
@@ -355,10 +355,55 @@ def test_sum_array(arrow_type):
 
     arr = pa.array([], type=arrow_type)
     assert arr.sum().as_py() is None  # noqa: E711
+    assert pc.sum(arr).as_py() is None  # noqa: E711
```

Comment:
You don't have to test both `arr.sum()` and `pc.sum(arr)` everytime since the former is just calling the latter. This will make the tests more readable and easier to maintain.

Reply:
Sure, but as it was, this test tested both `arr.sum()` and `pc.sum(arr)` for each of its subcases _except_ this one. The correct change would be to remove all those checks, which seemed like a larger change than making it consistent.

Reply:
Hobgoblins etc. ☺️ happy to remove these double checks as long as it’s the correct style moving forward!

Reply:
Done.